warp10/src/main/java/io/warp10/script/ScriptRunner.java

```diff
@@ -39,265 +68,232 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.locks.LockSupport;
 import java.util.zip.GZIPOutputStream;
 
-import org.apache.thrift.TException;
-import org.apache.thrift.TSerializer;
-import org.apache.thrift.protocol.TCompactProtocol;
-
-import com.geoxp.oss.CryptoHelper;
-import com.google.common.base.Charsets;
-import com.google.common.base.Preconditions;
-import com.google.common.primitives.Longs;
-import com.netflix.curator.framework.CuratorFramework;
-import com.netflix.curator.framework.CuratorFrameworkFactory;
-import com.netflix.curator.framework.recipes.leader.LeaderLatch;
-import com.netflix.curator.retry.RetryNTimes;
-
-import io.warp10.continuum.Configuration;
-import io.warp10.continuum.KafkaProducerPool;
-import io.warp10.continuum.KafkaSynchronizedConsumerPool;
-import io.warp10.continuum.TimeSource;
-import io.warp10.continuum.sensision.SensisionConstants;
-import io.warp10.continuum.store.Constants;
-import io.warp10.continuum.thrift.data.RunRequest;
-import io.warp10.crypto.CryptoUtils;
-import io.warp10.crypto.DummyKeyStore;
-import io.warp10.crypto.KeyStore;
-import io.warp10.crypto.OrderPreservingBase64;
-import io.warp10.crypto.SipHashInline;
-import io.warp10.sensision.Sensision;
-import kafka.javaapi.producer.Producer;
-import kafka.producer.KeyedMessage;
-import kafka.producer.ProducerConfig;
-
 /**
  * Periodically submit WarpScript scripts residing in subdirectories of the given root.
  * Greatly inspired by Sensision's own ScriptRunner
  */
 public class ScriptRunner extends Thread {
-  
+
+  private class NamedThreadFactory implements ThreadFactory {
+    private AtomicLong threadNumber = new AtomicLong();
+
+    @Override
+    public Thread newThread(Runnable r) {
+      return new Thread(r, "[Warp ScriptRunner Executor #" + threadNumber.getAndIncrement() + "]");
```

Comment:
Thread needs to be a daemon Thread.
Group needs to be set.
Priority needs to be set.

Check the DefaultThreadFactory from Executor.

Reply:
Thanks for the pointer, will fix.